panther_lights/src/controller_node.py

```diff
@@ -21,10 +20,89 @@
 
 
 class PantherAnimation:
+    ANIMATION_DEFAULT_PRIORITY = 3
+    ANIMATION_DEFAULT_TIMEOUT = 120
+
     front: Animation
     rear: Animation
-    interrupting = False
-    repeating = False
+    name: str = 'NAME_NOT_DEFINED'
+    priority: int = ANIMATION_DEFAULT_PRIORITY
+    timeout: float = ANIMATION_DEFAULT_TIMEOUT
+    repeating: bool = False
+
+    def __init__(self) -> None:
+        self._init_time = rospy.get_time()
+
+    @property
+    def init_time(self) -> float:
+        return self._init_time
+
+
+class AnimationsQueue:
+    def __init__(self, max_queue_size) -> None:
+        self._queue = []
+        self._max_queue_size = max_queue_size
+
+    def put(self, animation: PantherAnimation, put_front: int = False) -> None:
+        self.validate_queue()
+
+        if animation.priority == 1:
+            self.clear()
+
+        if len(self._queue) == self._max_queue_size:
+            rospy.logwarn(f'{rospy.get_name()} Animation queue overloaded')
+            return
+
+        if put_front:
+            self._queue.insert(0, animation)
+        else:
+            self._queue.append(animation)
+        self._queue.sort(key=self._get_priority)
+
+    def get(self) -> PantherAnimation:
+        return self._queue.pop(0)
+
+    def empty(self) -> bool:
+        return len(self._queue) == 0
+
+    def clear(self) -> None:
+        # clears queue except items with priority 1
+        remove_animation_list = []
+        for animation in self._queue:
+            if animation.priority > 1:
+                remove_animation_list.append(animation)
+        for animation in remove_animation_list:
+            self._queue.remove(animation)
+
+    def remove(self, animation) -> None:
+        if self.has_animation(animation):
+            self._queue.remove(animation)
+
+    def get_next_anim_priority(self) -> int:
+        if not self.empty():
+            return self._queue[0].priority
+        else:
+            return PantherAnimation.ANIMATION_DEFAULT_PRIORITY
+
+    def has_animation(self, animation: PantherAnimation) -> bool:
+        for anim in self._queue:
+            if anim == animation:
+                return True
+        return False
+
+    def validate_queue(self) -> None:
+        remove_animation_list = []
+        for animation in self._queue:
+            if rospy.get_time() - animation.init_time > animation.timeout:
+                remove_animation_list.append(animation)
+                rospy.logwarn(
+                    f'{rospy.get_name()} Timeout for animation: {animation.name}. Romoving from the queue'
+                )
+        for animation in remove_animation_list:
+            self._queue.remove(animation)
```

Comment:
```suggestion
t = rospy.get_time()
self._queue = [anim if t - anim.init_time < anim.timeout
               else f'{rospy.get_name()} Timeout for animation: {anim.name}. Removing from the queue'
               for anim in self._queue]
```
I am quite sure this abomination should be faster

Reply:
That won't work if we want to keep log info. In this case `else` in this form will put `str` into queue list